components/scream/src/share/io/output_manager.hpp

```diff
@@ -0,0 +1,271 @@
+ *
+ * --------------------------------------------------------------------------------
+ *  (2020-10-21) Aaron S. Donahue (LLNL)
+ */
+class OutputManager
+{
+public:
+  using output_type = AtmosphereOutput;
+
+  // Destructor
+  virtual ~OutputManager () = default;
+  // Constructors
+  OutputManager() = default;
+
+  OutputManager( const ekat::Comm& comm, const ekat::ParameterList& params,
+                 const std::shared_ptr<const FieldRepository<Real>>& repo,
+                 const std::shared_ptr<const GridsManager>& gm,
+                 const bool runtype_restart)
+  {
+    atm_comm            = comm;
+    m_params            = params;
+    param_set           = true;
+    m_device_field_repo = repo;
+    repo_set            = true;
+    m_grids_manager     = gm;
+    gm_set              = true;
+    m_runtype_restart   = runtype_restart;
+  }
+
+  OutputManager( const ekat::Comm& comm, const ekat::ParameterList& params,
+                 const std::shared_ptr<const FieldRepository<Real>>& repo,
+                 const std::shared_ptr<const GridsManager>& gm)
+  {
+    atm_comm            = comm;
+    m_params            = params;
+    param_set           = true;
+    m_device_field_repo = repo;
+    repo_set            = true;
+    m_grids_manager     = gm;
+    gm_set              = true;
+  }
+
+  void init();
+  void run(util::TimeStamp& current_ts);
+  void run(Real current_ts);
+  void finalize();
+
+  void new_output(const ekat::ParameterList& params);
+  void new_output(const ekat::ParameterList& params,const bool runtype_restart);
+  void set_comm(const ekat::Comm& comm) { atm_comm = comm; }
+  void set_params(const ekat::ParameterList& params) { m_params=params; param_set=true;}
+  void set_grids(const std::shared_ptr<const GridsManager>& gm) { m_grids_manager = gm; gm_set=true;}
+  void set_repo(const std::shared_ptr<const FieldRepository<Real>>& repo) { m_device_field_repo = repo; repo_set=true;}
+  void set_runtype_restart(const bool bval) { m_runtype_restart = bval; }
+  void make_restart_param_list(ekat::ParameterList& params);
+
+protected:
+  std::vector<std::shared_ptr<output_type>>    m_output_streams;
+  ekat::Comm                                   atm_comm;
+  ekat::Comm                                   pio_comm; 
+  ekat::ParameterList                          m_params;
+  std::shared_ptr<const FieldRepository<Real>> m_device_field_repo;
+  std::shared_ptr<const GridsManager>          m_grids_manager;
+
+  bool                                 param_set = false;
+  bool                                 gm_set    = false;
+  bool                                 repo_set  = false;
+  bool                        m_runtype_restart  = false;
+
+}; // class OutputManager
+/*===============================================================================================*/
+/* Short function to add a new output stream to the output manager.  By making this a stand-alone
```

Comment:
Not a big deal, but: a "standalone function" is typically different from a class method--it's just a function and not part of a class definition. I think you're trying to say that allowing someone to add a stream outside of the constructor is a useful thing, right?

Reply:
Also, consider sticking this comment above the declaration of the method within the class.